include/net/lwm2m.h

```diff
@@ -365,6 +366,99 @@ struct coap_block_context *lwm2m_firmware_get_block_context();
 #endif
 #endif
 
+
+#if defined(CONFIG_LWM2M_SWMGMT_OBJ_SUPPORT)
+
+/**
+ * @brief Set callback to handle software activation requests
+ *
+ * This callback will get called when the Activate execute on
+ * the corresponding object instance is called and the object is
+ * in the correct state.
+ *
+ * @param[in] obj_inst_id The instance number to set the callback for.
+ *
+ * @return 0 on success, otherwise a negative integer.
+ */
+int lwm2m_swmgmt_set_activate_cb(uint16_t obj_inst_id, lwm2m_engine_user_cb_t cb);
```

Comment:
I think `lwm2m_engine_execute_cb_t` would be more appropriate here. Same for other actions (deactivate/install/uninstall).

Reply:
Fixed